### PR TITLE
Search: Add result compression

### PR DESCRIPTION
**Pull request by pschoffer**
### Plugin Updated: Search

This change enables compression on HTTP requests towards Elastic Search.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [x] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- n/a This change has relevant documentation additions / updates (if applicable).
- [x] I've created a changelog description that aligns with the provided examples. 

## Steps to Test
1. Add a debug line `var_dump(['Response headers', $response['headers'] ]);`
2. Do any ES request e.g. `wp vip-search documents get post 1`
3. See the headers - `'content-encoding'` is not there
4. Apply the PR
5.  Do any ES request e.g. `wp vip-search documents get post 1`
6. Verify `'content-encoding'` header is present

### CODE REVIEW

File: search/includes/classes/class-search.php [hunk 1]
```diff
@@ -648,7 +648,14 @@ public function filter__ep_do_intercept_request( $request, $query, $args, $failu
 			$args['headers'] = [];
 		}
 
-		$args['headers'] = array_merge( $args['headers'], array( 'X-Client-Site-ID' => FILES_CLIENT_SITE_ID, 'X-Client-Env' => VIP_GO_ENV ) );
+		$args['headers'] = array_merge(
+			$args['headers'],
+			[
+				'X-Client-Site-ID' => FILES_CLIENT_SITE_ID,
+				'X-Client-Env' => VIP_GO_ENV,
+				'Accept-Encoding' => 'gzip, br',
```
Comment (rinatkhaziev) [inline search/includes/classes/class-search.php:656]:
```suggestion
				'Accept-Encoding' => 'gzip, deflate',
```

Brotli is currently not supported by the libcurl version we use:

```
curl --compressed https://httpbin.org/brotli
curl: (61) Unrecognized content encoding type. libcurl understands deflate, gzip content encodings.
```

**Review by rinatkhaziev — CHANGES_REQUESTED** [review]
Thanks for the PR, had no idea we don't do this already :)

Unfortunately `libcurl` we use is not compiled with brotli support, so let's swap that out to what we currently support.

**Comment by pschoffer** [comment]
Cool catch!

**Review by rinatkhaziev — APPROVED** [review]
:shipit:

**Review by nickdaugherty — APPROVED** [review]

**Comment by nickdaugherty** [comment]
r1416-stacks

**Comment by nickdaugherty** [comment]
@pschoffer I deployed this to staging, but i'm not 100% sure it's working as I'm not seeing the new header in requests when viewed through Debug Bar. It could be that it's a debug bar thing, but worth double checking.

**Comment by pschoffer** [comment]
I double checked today. It does seem to work when I print out headers directly in code. I got:
```
 string 'HTTP/1.1 200 OK
Server: nginx
Date: Tue, 02 Mar 2021 10:33:09 GMT
Content-Type: application/json; charset=UTF-8
Transfer-Encoding: chunked
Connection: close
Vary: Accept-Encoding
Content-Encoding: gzip
```

So that looks good. I still see only a subset of the request headers without the `Accept-Encoding` among others. So I guess it is filtered somewhere in debug bar.

**Comment by nickdaugherty** [comment]
Great, thanks for checking @pschoffer !

**Pull request merged**